DirectProgramming/DPC++FPGA/Tutorials/Tools/experimental/platform_designer/Readme.md

```diff
@@ -0,0 +1,274 @@
+
+   2. Set the top-level entity to be `add` to make project management easier.
+
+      ![](Readme.md.assets/quartus_new_project.png)
+
+   4. Choose **Empty Project** when prompted to select a project type.
+
+   5. Add the source file `add.v` and `jtag.sdc` to the design when the wizard prompts you.
+
+      ![](Readme.md.assets/add-files.png)
+
+   4. Make sure you choose an appropriate device. See **Board-specific Considerations** above.
+
+3. Copy the oneAPI-generated IP to the Intel Quartus® Prime project. This design uses host pipes, which generates additional internal SYCL kernels. The `fpga_export` build target uses the `-fsycl-device-code-split=per_kernel` flag to separate these additional kernels from your kernel, but these kernels will have their own reports and associated RTL. You will therefore need to hunt for the `.prj_X` directory that contains the IP you want to use in your design.
+
+   You can identify the correct `.prj_X` folder by looking for the one that contains a `*_di_inst.v` whose interfaces match your kernel. For example, in this project, `add_xample.fpga_ip.prj_1` is the correct `.prj_x` directory, because `add_example_fpga_ip_1_di_inst.v` contains only a CSR Agent interface in addition to the clock/reset signals:
+   
+   ```
+   add_example_fpga_ip_1_di add_example_fpga_ip_1_di_inst (
+     // Interface: clock (clock end)
+     .clock                          ( ), // 1-bit clk input
+     // Interface: clock2x (clock end)
+     .clock2x                        ( ), // 1-bit clk input
+     // Interface: resetn (conduit end)
+     .resetn                         ( ), // 1-bit data input
+     // Interface: device_exception_bus (conduit end)
+     .device_exception_bus           ( ), // 64-bit data output
+     // Interface: kernel_irqs (interrupt end)
+     .kernel_irqs                    ( ), // 1-bit irq output
+     // Interface: csr_ring_root_avs (avalon end)
+     .csr_ring_root_avs_read         ( ), // 1-bit read input
+     .csr_ring_root_avs_readdata     ( ), // 64-bit readdata output
+     .csr_ring_root_avs_readdatavalid( ), // 1-bit readdatavalid output
+     .csr_ring_root_avs_write        ( ), // 1-bit write input
+     .csr_ring_root_avs_writedata    ( ), // 64-bit writedata input
+     .csr_ring_root_avs_address      ( ), // 5-bit address input
+     .csr_ring_root_avs_byteenable   ( ), // 8-bit byteenable input
+     .csr_ring_root_avs_waitrequest  ( )  // 1-bit waitrequest output
+   );
+   ```
+
+   Linux:
+
+   ```
+   $> cd .. # navigate to project root if not there already
+   $> cp -r add-oneapi/build/add.fpga_ip_export.prj_1/ add-quartus/
+   ```
+
+   Windows:
+
+   ```
+   > cd .. # navigate to project root if not there already
+   > xcopy add-oneapi\build\add.fpga_ip_export.prj_1\ add-quartus\add.fpga_ip_export.prj_1 /e /s /i
+   ```
+
+4. Correct the generated `_hw.tcl` file by running the `*_di_hw_tcl_adjustment_script.py` script in the .prj directory.
```

Comment:
Huh?  Why does the _hw.tcl need to be 'corrected'?  Does the oneAPI compiler spit out both the _hw.tcl and the 'adjustment' script?  Why doesn't it just spit out the _hw.tcl already 'adjusted'?

I think you should explain what 'correction' is being applied here.

Reply:
this is a 2023.0 bug that is fixed in 2023.1. be glad you get a 'correction' script... we almost had to manually edit 10 lines of text.

I don't want to go into detail because this issue will be fixed in 2023.1, and we don't plan to market IPA until then.